tests/regression/test_expressions.py

```diff
@@ -169,10 +169,60 @@ def test_tensor_expressions(expr, tfunctions):
     assert eval(expr)
 
 
-@pytest.mark.parametrize('expr', common_tests)
-def test_mixed_expressions(expr, mfunctions):
+def test_mixed_expressions(mfunctions):
     f, one, two = mfunctions
-    assert eval(expr)
+
+    f.sub(0).assign(one.sub(0))
+    assert evaluate(f.dat.data, (1, 0))
+    f.assign(0)
+
+    f.sub(1).assign(one.sub(1))
+    assert evaluate(f.dat.data, (0, 1))
+    f.assign(0)
+
+    two.sub(0).assign(one.sub(0))
+    assert evaluate(two.dat.data, (1, 2))
+    two.assign(2)
+
+    two.sub(1).assign(one.sub(1))
+    assert evaluate(two.dat.data, (2, 1))
+    two.assign(2)
+
+    two.sub(0).assign(one.sub(0) + two.sub(0))
+    assert evaluate(two.dat.data, (3, 2))
+    two.assign(2)
+
+    two.sub(1).assign(two.sub(1) - one.sub(1))
+    assert evaluate(two.dat.data, (2, 1))
+    two.assign(2)
+
+    one0 = one.sub(0)
+    one0 += one.sub(0)
+    assert evaluate(one.dat.data, (2, 1))
+    one.assign(1)
+
+    one1 = one.sub(1)
+    one1 -= one.sub(1)
+    assert evaluate(one.dat.data, (1, 0))
+
+
+def test_mixed_expressions_indexed_fs(msfunctions):
+    f, one, two = msfunctions
+
+    f.sub(0).assign(one)
+    assert evaluate(f.dat.data, (1, 0))
+    f.assign(0)
+
+    f.sub(1).assign(two)
+    assert evaluate(f.dat.data, (0, 2))
+    f.sub(0).assign(one)
+    assert evaluate(f.dat.data, (1, 2))
+
+    one.assign(2*f.sub(0) + 1)
+    assert evaluate(one.dat.data, 3)
+
+    two += f.sub(1)
+    assert evaluate(two.dat.data, 4)
 
```

Comment:
I think @JHopeCollins might have some more tests to contribute, but these can be added at a later date